lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp

```diff
@@ -1875,17 +1894,22 @@ TEST_F(
   EXPECT_EQ(mme_state_p->nb_default_eps_bearers, 1);
   EXPECT_EQ(mme_state_p->nb_ue_idle, 0);
 
-  // Send activate dedicated bearer request mimicing SPGW
-  send_s11_create_bearer_req();
+  // Send activate dedicated bearer request for lbi 5 mimicing SPGW
+  send_s11_create_bearer_req(5);
   EXPECT_CALL(*s1ap_handler, s1ap_generate_s1ap_e_rab_setup_req()).Times(1);
 
   // Send ERAB Setup Response mimicing S1AP
-  send_erab_setup_rsp();
+  send_erab_setup_rsp(6);
 
   // Wait for DL NAS Transport up to retransmission limit.
   // The first transmission was piggybacked on ERAB Setup Request
   for (int i = 1; i < NAS_RETX_LIMIT; ++i) {
     cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+    cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+    cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+    cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+    cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+    cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
```

Comment:
what are these extra waits for?

Reply:
T3485 timer is set to 10ms in nas_config_timer_reinit() which works for timer expiry scenario but for a normal case (TestNormalTauReqWithEpsBearerCtxStsInactiveDedBearer and TestNormalTauReqWithEpsBearerCtxStsInactiveDefBearer) when secondary PDN connect message is sent from UE, MME will send create session req to spgw, get back create session rsp from spgw, receive erab setup rsp and then receive activate default eps bearer accept. For all these events to happen 10ms is less. After a lot of trial and error I could see fix the T3485 value to 5000ms. Because of this I had to increase the wait time here to make sure T3485 timer expires as per the TC requirement

Reply:
check out https://github.com/magma/magma/pull/10747. I tested with your PR with that change and removed all these extras with no issues.